templates/vue/src/components/Lightbox/media/ActivityMedia/AudioRecorder.vue

```diff
@@ -95,29 +118,38 @@ export default {
       return `${hours}:${minutes}:${sec}`
     },
     hasPrevious() {
-      return (
-        this.question.entries &&
-        this.question.entries.audioId &&
-        this.question.entries.audioId.length > 0
-      )
+      let answers = this.getAnswers(this.node.id, this.question.id)
+      return answers.audio && answers.audio.url
     },
     states() {
+      // NOTE: DONE state is used to determine if the audio already exists in the backend
```

Comment:
Rather than this node, why not rename "DONE" to "SAVED"? I also think "WAITING_TO_BE_SENT" can be renamed to "UNSAVED" and "WAIT" can be renamed "LOADING". I also suggest reordering this array to put the values in a more intuitive order (E.g. SAVED should be the last state).